app/components/shared/avatar-uploader.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  type OutputFileEntry,
+  type UploadCtxProvider,
+} from "@uploadcare/blocks"
+import { useEffect, useRef } from "react"
+
+import {
+  LRConfig,
+  LRFileUploaderRegular,
+  defaultLRConfig,
+  type UploaderConfigProps,
+} from "../libs/uploader-uploadcare"
+
+export interface UploaderWithPreviewProps extends UploaderConfigProps {
+  setFile: (files: OutputFileEntry | null) => void
+}
+
+export function UploaderWithProvider({
+  pubkey = "demopublickey",
+  contextName = "my-uploader",
+  theme = "light",
+  config = defaultLRConfig,
+  setFile,
+  ...props
+}: UploaderWithPreviewProps) {
+  const ctxProviderRef = useRef<
+    typeof UploadCtxProvider.prototype & UploadCtxProvider
+  >(null)
+
+  useEffect(() => {
+    const handleUploadEvent = (e: CustomEvent<OutputFileEntry[]>) => {
+      if (e.detail) {
+        setFile(e.detail[0] ?? null)
+      }
+    }
+
+    ctxProviderRef.current?.addEventListener("data-output", handleUploadEvent)
```

Comment:
Please check this part , currently I still got typescript error, even though I  already follow the official example : https://github.com/uploadcare/blocks-examples/blob/main/examples/react-uploader/src/components/FileUploader/FileUploader.tsx#L21

Reply:
the TypeScript errors can be ignored, use ts-ignore, because originally the Uploadcare component and its parts are not well supported with TypeScript yet